homeassistant/components/folder_watcher.py

```diff
@@ -0,0 +1,111 @@
+"""
+Component for monitoring activity on a folder.
+
+For more details about this platform, refer to the documentation at
+https://home-assistant.io/components/folder_watcher/
+"""
+import os
+import logging
+import voluptuous as vol
+from homeassistant.const import (
+    EVENT_HOMEASSISTANT_START, EVENT_HOMEASSISTANT_STOP)
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['watchdog==0.8.3']
+_LOGGER = logging.getLogger(__name__)
+
+CONF_FOLDER = 'folder'
+CONF_PATTERNS = 'patterns'
+CONF_WATCHERS = 'watchers'
+DEFAULT_PATTERN = '*'
+DOMAIN = "folder_watcher"
+EVENT_TYPE = "event_type"
+FILE = 'file'
+FOLDER = 'folder'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.All(cv.ensure_list, [vol.Schema({
+        vol.Required(CONF_FOLDER): cv.isdir,
+        vol.Optional(CONF_PATTERNS, default=[DEFAULT_PATTERN]):
+            vol.All(cv.ensure_list, [cv.string]),
+    })])
+}, extra=vol.ALLOW_EXTRA)
+
+
+def setup(hass, config):
+    """Set up the folder watcher."""
+    conf = config[DOMAIN]
+    for watcher in conf:
+        path = watcher[CONF_FOLDER]
+        patterns = watcher[CONF_PATTERNS]
+        if not hass.config.is_allowed_path(path):
+            _LOGGER.error("folder %s is not valid or allowed", path)
+            continue
```

Comment:
Hmmm, should we maybe consider returning `False` (so creating a persistent notification saying we weren't able to set up)? I mean few users actually look through the logs and so maybe some would not notice their folder watcher not being set up. 

With this integration it would be especially hard to debug the issue as a user (if you're not looking at the logs) because you don't have any immediate visual feedback that setup worked or not. For example if you create a folder watcher for an unallowed path, you would just not receive the events. That might confuse some people.

Is there any use case for breaking kind of silently anyway?

Reply:
Adopting the `False `